src/table/src/metadata.rs

```diff
@@ -588,6 +597,95 @@ impl TableMeta {
         Ok(meta_builder)
     }
 
+    fn change_column_fulltext(
+        &self,
+        table_name: &str,
+        column_name: &String,
+        options: HashMap<String, String>,
+    ) -> Result<TableMetaBuilder> {
+        let table_schema = &self.schema;
+        let mut meta_builder = self.new_meta_builder();
+
+        let index = table_schema
+            .column_index_by_name(column_name)
+            .with_context(|| error::ColumnNotExistsSnafu {
+                column_name,
+                table_name,
+            })?;
+        let column = &table_schema.column_schemas()[index];
+
+        let mut fulltext = if let Ok(Some(f)) = column.fulltext_options() {
+            f
+        } else {
+            FulltextOptions::default()
+        };
+
+        if let Some(analyzer) = options.get(COLUMN_FULLTEXT_OPT_KEY_ANALYZER) {
+            match analyzer.to_ascii_lowercase().as_str() {
+                "english" => {
+                    fulltext.enable = true;
+                    fulltext.analyzer = FulltextAnalyzer::English;
+                }
+                "chinese" => {
+                    fulltext.enable = true;
+                    fulltext.analyzer = FulltextAnalyzer::Chinese;
+                }
+                _ => {
+                    return error::InvalidFulltextOptionsSnafu { column_name }.fail()?;
+                }
+            }
+        }
+        if let Some(case_sensitive) = options.get(COLUMN_FULLTEXT_OPT_KEY_CASE_SENSITIVE) {
+            match case_sensitive.to_ascii_lowercase().as_str() {
+                "true" => {
+                    fulltext.enable = true;
+                    fulltext.case_sensitive = true;
+                }
+                "false" => {
+                    fulltext.enable = true;
+                    fulltext.case_sensitive = false;
+                }
+                _ => {
+                    return error::InvalidFulltextOptionsSnafu { column_name }.fail()?;
+                }
+            }
+        }
+
+        let columns: Vec<_> = table_schema
+            .column_schemas()
+            .iter()
+            .cloned()
+            .map(|mut column| {
+                if column.name == *column_name {
+                    column = column
+                        .clone()
+                        .with_fulltext_options(fulltext.clone())
+                        .unwrap();
+                }
+                column
+            })
+            .collect();
```

Comment:
1. Avoid cloning
2. Handle the error
```rust
let column = table_schema
            .column_schemas()
            .iter_mut()
            .find(|column| column.name = column_name);
        column
            .context(ColumnNotFound {})
            .set_fulltext_options(fulltext)?;
```

Reply:
To use `iter_mut()` we have to borrow table_schema  as `&mut self.schema`, which is conflict with `self.new_meta_builder()`.